### ISSUE TITLE
Feature Request: Read process interleaved with stream assignment

### ISSUE THREAD

**Issue by ad-si** [fpco/typed-process#61]
It would be cool to have a function a la

```hs
readProcessTimed
  :: MonadIO m	  
  => ProcessConfig stdin stdoutIgnored stderrIgnored	  
  -> m (ExitCode, [(ByteString, ByteString, ByteString)])
```

which would results in something like this:

```hs
( ExitSuccess,
    [ ( "2022-10-29T15:20:12.168Z", "stdout", "Hello World!" ) ] 
    , ( "2022-10-29T15:20:12.175Z", "stderr", "ERROR: World not available" ) ] 
    ]
)
```

- There should probably be a version with and without timestamp (or make it configurable)
- It should probably use a more elaborate data type than a tuple of strings

**Comment by tomjaguarpaw** [comment]
I don't think I follow. What do the timestamps refer to here? The first time some data was available on that stream?

**Comment by ad-si** [comment]
Well, that's up for debate I guess, but Google Cloud Functions for example uses a timestamp for each line (terminated by `\n`). I guess it's the moment the \n appears on the stream.

**Comment by tomjaguarpaw** [comment]
Personally I'd say that kind of thing belongs in a higher level library and that `typed-process` is too low-level for that. @snoyberg makes the final decision though.

**Comment by snoyberg** [comment]
I agree with @tomjaguarpaw's assessment.

**Comment by ad-si** [comment]
Ok, but even for a version without timestamps, which tags where each chunk came from?
```hs
( ExitSuccess,
    [ Stdout "Hello World!"
    , Stderr "ERROR: World not available"
    ]
)
```

**Comment by tomjaguarpaw** [comment]
Yes, even for that. How about this: you write something that does that, use it happily in production for a year, and then come back here with a PR saying "this is rock solid in production and really useful, would you like to incorporate it?". Then we're more likely to add it to the library.
